src/exportAuthRecord.ts

```diff
@@ -0,0 +1,143 @@
+    let tenantFromArg: string | undefined = undefined;
+    try {
+        // This handles the case if an error is thrown, if the configuration is not registered by any extension
+        tenantFromArg = vscode.workspace.getConfiguration().get<string>('@azure.argTenant');
+    } catch (err) {
+        // If the configuration is not found, ignore and proceed
+        ext.outputChannel.appendLine('No @azure.argTenant configuration found. Proceeding without tenant override.');
+    }
+    if (tenantFromArg) {
+        return tenantFromArg;
+    }
+    return extractTenantIdFromIdToken(session, context);
+}
+
+// Helper to extract tenantId (tid) from the idToken of a VS Code authentication session (For MS Auth, this will be present).
+function extractTenantIdFromIdToken(session: unknown, context?: IActionContext): string | undefined {
+    const idToken = (session as { idToken?: unknown }).idToken;
+    if (typeof idToken === 'string') {
+        const parts = idToken.split('.');
+        if (parts.length === 3) {
+            try {
+                const payloadStr = Buffer.from(parts[1], 'base64').toString('utf8');
+                const payload = JSON.parse(payloadStr) as Record<string, unknown>;
+                if (typeof payload.tid === 'string') {
+                    return payload.tid;
+                }
+            } catch (e) {
+                ext.outputChannel.appendLine('Failed to parse idToken for tenantId.');
+                if (context) {
+                    context.telemetry.properties.result = 'Failed';
+                }
+            }
+        }
+    }
+    return undefined;
+}
+
+// Helper to persist the authentication record to the user's .azure directory
+async function persistAuthRecord(authRecord: Record<string, unknown>): Promise<void> {
+    // Write to a well-known location in .azure home directory (handle both .azure and .Azure for cross-platform compatibility)
+    // If either exists, use it; otherwise, create .azure
+    // This ensures a consistent, well-known location for other tools and applications to store and retrieve authentication records.
+    // Allows tools to write auth records to a central location for other apps to read from.
+    // Didn't use GlobalStorage path, as that brings added risk of regressions for auth record location which is used by external tools and apps, if the location changes in the future.
+    const homeDir = os.homedir();
+    const azureDir = path.join(homeDir, '.azure');
+    const altAzureDir = path.join(homeDir, '.Azure');
+    let baseAzureDir: string;
+    if (await fs.pathExists(azureDir)) {
+        baseAzureDir = azureDir;
+    } else if (await fs.pathExists(altAzureDir)) {
+        baseAzureDir = altAzureDir;
+    } else {
+        baseAzureDir = azureDir;
+        await fs.ensureDir(baseAzureDir);
+    }
+    const authDir = path.join(baseAzureDir, 'auth-records', 'ms-azuretools.vscode-azureresourcegroups');
```

Comment:
Unless there's something that requires it, I suggest removing the auth-records directory from this path. The JSON file name already makes it clear that's what the data represents.

Reply:
no strong opinion, 
The idea was to centralize the auth-records, if and as we get other tools to write their auth record as well. 
.azure/auth-records/azpwsh/
.azure/auth-records/ms-azuretools.vscode-azureresourcegroups

Reply:
removed.